websites/J/JoJo's Bizarre Encyclopedia/presence.ts

```diff
@@ -2,10 +2,15 @@ const presence = new Presence({
 	clientId: "1007688410875564062",
 });
 
-presence.on("UpdateData", () => {
+presence.on("UpdateData", async () => {
 	let presenceData: PresenceData = {};
 
-	if (window.location.href.toLowerCase().includes("jojowiki.com")) {
+	const showButtons = await presence.getSetting("showButtons"),
+		{ hostname, pathname, origin, href, searchParams } = new URL(
```

Comment:
You can use `document.location.hostname` `document.location.pathname` `document.location.origin` `document.location.href` instead
For `searchParams` you may just use `const searchParams = new URLSearchParams(document.location.search)`

Reply:
I'm not sure if I get what you mean completely, but if you mean to use document.location instead of the URL object I see no purpose in this, as the URL object is already created anyways, and declaring URLSearchParams separately seems unnecessary, unless I'm missing something.